emu/src/arm/arm7tdmi.rs

```diff
@@ -50,18 +51,18 @@ pub struct Arm7tdmi {
 
     pub cpsr: Psr,
     pub registers: Registers,
+
+    pub register_bank: RegisterBank,
 }
 
 impl Default for Arm7tdmi {
     fn default() -> Self {
-        // SPSR are initalized with the value related to their respective
-        // mode (5 least significant bits).
-        // e.g., FIQ SPSR has mode 0b10001, IRQ 0b10010, etc...
         Self {
             rom: Arc::new(Mutex::new(Vec::default())),
             memory: Arc::new(Mutex::new(InternalMemory::default())),
             cpsr: Psr::from(Mode::User), // FIXME: Starting as User? Not sure
             registers: Registers::default(),
+            register_bank: RegisterBank::default(),
```

Comment:
Maybe we can `..Default::default()`

Reply:
I tried but then it's recursive :(

Reply:
Oh yes yes, we are implementing default here :)